src/System.Management.Automation/namespaces/LocationGlobber.cs

```diff
@@ -1665,9 +1665,16 @@ internal static bool IsAbsolutePath(string path)
 
                 if (index > 0)
                 {
-                    // We must have a drive specified
-
-                    result = true;
+                    int separator = path.IndexOf(StringLiterals.DefaultPathSeparator, 0, index);
```

Comment:
Based on the [documentation](https://msdn.microsoft.com/en-us/library/ms131434(v=vs.110).aspx) this does not look correct to me. It searches the string starting from the beginning though index count of characters for the matching separator. "C:\something.txt" Will always fail because the separator never matches the first character of the string ("C").

I think you might need:
```c#
separator = path.IndexOf(StringLiterals.AlternatePathSeparator, index, 1); 
```
This checks the character immediately after the index. But if you are going to do that, you might as well directly test index+1.

Reply:
`index` points to location of the colon.  So this test is to determine if there is a path separator between the beginning and the colon.  So it should match `./foo:bar` but not match `foo:bar` where the former is a file/directory and the latter is a drive path.  One optimization I can see is I only need to search up to `index-1` since I already know there is a colon at `index`.